src/vs/workbench/parts/tasks/electron-browser/task.contribution.ts

```diff
@@ -769,7 +769,10 @@ class TaskService extends Disposable implements ITaskService {
 	}
 
 	public getRecentlyUsedTasks(): LinkedMap<string, string> {
+		const quickOpenHistoryLimit = this.configurationService.getValue<number>('workbench.tasks.quickOpen.history');
```

Comment:
I think we should do the size limiting when we create the recently used task cache here: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/tasks/electron-browser/task.contribution.ts#L771

Instead of using LinkedMap we should use LRUCache with a size limit and resize the cache when the setting changes.

Reply:
Yeah, using the LRU map would be better for the cause you pointed to above.

Reply:
Okay, I've refactored the PR with your suggestions and put the LRUCache in place. Have a look.